### ISSUE TITLE
Crash when crafting blasting powder

### ISSUE THREAD

**Issue by copygirl** [anegostudios/VintageStory-Issues#7]
> [VintagestoryCrash.txt](https://www.vintagestory.at/applications/core/interface/file/attachment.php?id=1202)
> 
> Single player world survival no mods

~ Don Rudski [on the forums](https://www.vintagestory.at/forums/topic/422-crash-crafting-blasting-powder-152/?do=findComment&comment=2086)

```
2018-03-18 5:39:29 PM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.CraftingRecipeIngredient.Clone()
   at Vintagestory.API.Common.GridRecipe.ConsumeInputShapeLess(IWorldAccessor world, IPlayer byPlayer, IItemSlot[] inputSlots)
   at _pU4W8AID8egAOqbOUSPWtLjlpoG._LyXzQdaVvzSo0F2bEpfxsaRsrPm()
   at _dPnVZxuzVo9zTepb8bQLbovABOD._mf5mBYp5Dc4L94ExFtpJPxx6U0i(IItemSlot , ItemStackMoveOperation& )
   at Vintagestory.API.Common.ItemSlotOutput.ActivateSlot(ItemSlot sourceSlot, ItemStackMoveOperation& op)
   at Vintagestory.API.Common.InventoryBase.ActivateSlot(Int32 slotId, IItemSlot sourceSlot, ItemStackMoveOperation& op)
   at _eIXvGyY93zPWhP6CvNwVVgvbawO._Lu1gJgMzkjJ82msCEBOXlJe9ASk(_ir11PTuRpE1aeYwMlNioYG2kBeX , Int32 , EnumMouseButton , Boolean , Boolean , Boolean )
   at _eIXvGyY93zPWhP6CvNwVVgvbawO._6qKqgzRf8BXOrGYOovwNRG9Z7aG(_ir11PTuRpE1aeYwMlNioYG2kBeX , MouseEvent )
   at _Vrnv6thBPy1954jaoG6PSwrR0NG._CJ72SVkxJwxCA1frA5OqhOsLkpe(_ir11PTuRpE1aeYwMlNioYG2kBeX , MouseEvent )
   at _Z0rMF0Dnl5mhZmH0nXamPE2thwD._CJ72SVkxJwxCA1frA5OqhOsLkpe(_ir11PTuRpE1aeYwMlNioYG2kBeX , MouseEvent )
   at _KMLWua1Zvn3hcr6cvXfFfHq5MYB._CJ72SVkxJwxCA1frA5OqhOsLkpe(MouseEvent )
   at _MrQsu2GuJMLQvRd30L2qnPMhJoh._CJ72SVkxJwxCA1frA5OqhOsLkpe(MouseEvent )
   at _1Xag6se2FILA9eGXXRFFOHwGBcb._CJ72SVkxJwxCA1frA5OqhOsLkpe(MouseEvent )
   at _lpa2wezMUmPC9SYeqyldCPi5ykI._mJvNcPMAIwehIVIWxp5tJn5z6cd(Object , MouseButtonEventArgs )
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.Platform.Windows.WinGLNative.WindowProcedure(IntPtr handle, WindowMessage message, IntPtr wParam, IntPtr lParam)
   at OpenTK.Platform.Windows.Functions.DispatchMessage(MSG& msg)
   at OpenTK.Platform.Windows.WinGLNative.ProcessEvents()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _Tv2OyERsF5nbcsgRKs9eUwKhqvQ._PKov0Of2By4G8K7ADCC7QTQGmRJ(_mkP9VlL7dCVktBhZ5FljAa2Dtjn , String[] )
   at _KvvPcWG9QIDQlkC88RkzBikZZNB._PKov0Of2By4G8K7ADCC7QTQGmRJ(ThreadStart )
```

**Comment by copygirl** [comment]
donald12999 on Discord has a related issue when crafting planks.

> **donald12999:** we tracking that planks cause a crash?
> **donald12999:** And is there a work around?
> **donald12999:** ...
> **Tyron:** drop me crash logs
> **Tyron:** then it will be in tomorrows bugfix release, probably
> **donald12999:** lso [VintagestoryCrash.txt](https://cdn.discordapp.com/attachments/334233095374897152/425051420799270942/VintagestoryCrash.txt)
> **donald12999:** ive tried a couple types of logs too
> **donald12999:** both shift click and manual cause the crash

```
-------------------------------
3/18/2018 5:52:05 PM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.CraftingRecipeIngredient.Clone()
   at Vintagestory.API.Common.GridRecipe.ConsumeInputShapeLess(IWorldAccessor world, IPlayer byPlayer, IItemSlot[] inputSlots)
   at _pU4W8AID8egAOqbOUSPWtLjlpoG._LyXzQdaVvzSo0F2bEpfxsaRsrPm()
   at _dPnVZxuzVo9zTepb8bQLbovABOD._sKTCWCdaCzr3BOBkQ4voBQnRm8o(IItemSlot , ItemStackMoveOperation& )
   at _Rrg61M4V1Vj2PIrdS3VEaf1VEyD.TryTransferItemFrom(IItemSlot sourceSlot, ItemStackMoveOperation& op, Boolean onlyPlayerInventory, Boolean slotNotifyEffect)
   at _eIXvGyY93zPWhP6CvNwVVgvbawO._Lu1gJgMzkjJ82msCEBOXlJe9ASk(_ir11PTuRpE1aeYwMlNioYG2kBeX , Int32 , EnumMouseButton , Boolean , Boolean , Boolean )
   at _eIXvGyY93zPWhP6CvNwVVgvbawO._6qKqgzRf8BXOrGYOovwNRG9Z7aG(_ir11PTuRpE1aeYwMlNioYG2kBeX , MouseEvent )
   at _Vrnv6thBPy1954jaoG6PSwrR0NG._CJ72SVkxJwxCA1frA5OqhOsLkpe(_ir11PTuRpE1aeYwMlNioYG2kBeX , MouseEvent )
   at _Z0rMF0Dnl5mhZmH0nXamPE2thwD._CJ72SVkxJwxCA1frA5OqhOsLkpe(_ir11PTuRpE1aeYwMlNioYG2kBeX , MouseEvent )
   at _KMLWua1Zvn3hcr6cvXfFfHq5MYB._CJ72SVkxJwxCA1frA5OqhOsLkpe(MouseEvent )
   at _MrQsu2GuJMLQvRd30L2qnPMhJoh._CJ72SVkxJwxCA1frA5OqhOsLkpe(MouseEvent )
   at _1Xag6se2FILA9eGXXRFFOHwGBcb._CJ72SVkxJwxCA1frA5OqhOsLkpe(MouseEvent )
   at _lpa2wezMUmPC9SYeqyldCPi5ykI._mJvNcPMAIwehIVIWxp5tJn5z6cd(Object , MouseButtonEventArgs )
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.Platform.Windows.WinGLNative.WindowProcedure(IntPtr handle, WindowMessage message, IntPtr wParam, IntPtr lParam)
   at OpenTK.Platform.Windows.Functions.DispatchMessage(MSG& msg)
   at OpenTK.Platform.Windows.WinGLNative.ProcessEvents()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _Tv2OyERsF5nbcsgRKs9eUwKhqvQ._PKov0Of2By4G8K7ADCC7QTQGmRJ(_mkP9VlL7dCVktBhZ5FljAa2Dtjn , String[] )
   at _KvvPcWG9QIDQlkC88RkzBikZZNB._PKov0Of2By4G8K7ADCC7QTQGmRJ(ThreadStart )
-------------------------------
3/18/2018 6:01:51 PM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at _2NIJUOvZuhsZJHwssaR4A9MeQQd._evQawCNQwidlgfLJr6UALkdxogBb(_YNbrYWrfHwoFb935Yyu9Lq954vg )
   at _OIvMSh21UgCkAuQfMesEJXrxu7A._W7pFANs1eFrLvcNwzoTm2ed0AaH(_1Xag6se2FILA9eGXXRFFOHwGBcb , Single )
   at _1Xag6se2FILA9eGXXRFFOHwGBcb._l49OQFCZAUy2gy55LtpANhlramM(Single )
   at _7frinaibFVrKQRnO7A7QhU2Z9Np._7dpc4nVDU3eu39XW5zQM8hvxaGq(Single )
   at _PEvCJqwBnvvamlPCwiTDs2jyp0t._vrRiMNCCuqWbOC92sGUK762eBEv(Single )
   at _PEvCJqwBnvvamlPCwiTDs2jyp0t._bJCcIXsEn7jCQIBmjq5EGMetZEWA(Single )
   at _lpa2wezMUmPC9SYeqyldCPi5ykI._lalnwkS7TIV3zPLuBnmGOGMzkcA(Object , FrameEventArgs )
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _Tv2OyERsF5nbcsgRKs9eUwKhqvQ._PKov0Of2By4G8K7ADCC7QTQGmRJ(_mkP9VlL7dCVktBhZ5FljAa2Dtjn , String[] )
   at _KvvPcWG9QIDQlkC88RkzBikZZNB._PKov0Of2By4G8K7ADCC7QTQGmRJ(ThreadStart )
-------------------------------
```

**Comment by tyronx** [comment]
Ok, no idea why this suddenly appeared but should be fixed for 1.5.2.1, please reopen if the issue persists.
